textdb/textdb-perftest/src/main/java/edu/uci/ics/textdb/perftest/excelsink/ExcelSink.java

```diff
@@ -0,0 +1,164 @@
+        this.inputOperator = inputOperator;
+    }
+    
+    public IOperator getInputOperator() {
+        return this.inputOperator;
+    }
+
+    @Override
+    public Schema getOutputSchema() {
+        return outputSchema;
+    }
+
+    @Override
+    public void open() throws TextDBException{
+        if (isOpen) {
+            return;
+        }     
+        inputOperator.open();
+        inputSchema = inputOperator.getOutputSchema();
+        outputSchema = Utils.removeAttributeFromSchema(inputSchema, SchemaConstants._ID, SchemaConstants.PAYLOAD);
+        try{
+            wb = new XSSFWorkbook();
+            DateFormat df = new SimpleDateFormat("YYYYMMDD_HH_mm_ss_SSSS");
+            Date dateobj = new Date();
+            fileName = df.format(dateobj) + ".xlsx";
+        	fileOut = new FileOutputStream(excelIndexDirectory + fileName);
+        	sheet = wb.createSheet("new sheet");
+        	row = sheet.createRow((short)0);
+        	List<String> attributeNames = outputSchema.getAttributeNames();
+        	//TODO::if attribute is a list, we cannot give it a fixed column size
+        	for(int i = 0; i < attributeNames.size(); i++){
+        		String attributeName = attributeNames.get(i);
+            	row.createCell(i).setCellValue(attributeName);
+        	}
+            isOpen = true;
+        }catch(Exception e){
+        	System.out.println("ExcelSink::open  "+e.getMessage());
+        }
+    }
+
+    @Override
+    public void processTuples() throws TextDBException {
+        return;
+    }
+    
+    @Override
+    public Tuple getNextTuple() throws TextDBException {
+        Tuple tuple = inputOperator.getNextTuple();
+        if (tuple == null) {
+            return null;
+        }
+        return Utils.removeFields(tuple, SchemaConstants._ID, SchemaConstants.PAYLOAD);
+    }
+
+    @Override
+    public void close() throws TextDBException {
+        if (isOpen) {
+        	inputOperator.close();
+            try {
+                wb.write(fileOut);
+    			fileOut.close();
+                isOpen = false; 
+    		} catch (IOException e) {
+            	System.out.println("ExcelSink::open  "+e.getMessage());
+    		}
+        }
+    }
+
+    /**
+     * Collects ALL the tuples to an in-memory list, but also add all tuples into the excel file
+     * 
+     * @return a list of tuples
+     * @throws TextDBException
+     */
+    public List<Tuple> collectAllTuples() throws TextDBException {
+        ArrayList<Tuple> results = new ArrayList<>();
+        Tuple tuple;
+        while ((tuple = inputOperator.getNextTuple()) != null) {
+            results.add(Utils.removeFields(tuple, SchemaConstants._ID, SchemaConstants.PAYLOAD));
+        }
+        int count = 1;
+        for(Tuple tp: results){
+        	row = sheet.createRow(count++);
+        	List<IField> fields = tp.getFields();
+        	for(int i = 0; i < fields.size(); i++){
+        		row.createCell(i).setCellValue(fields.get(i).getValue().toString());
```

Comment:
Don't `toString` everything. If the value is `int` or `double`. We want the value in excel file also to be a number, not a string